sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -921,12 +924,18 @@ class Analyzer(
         failAnalysis("Invalid usage of '*' in explode/json_tuple/UDTF")
 
       // To resolve duplicate expression IDs for Join and Intersect
-      case j @ Join(left, right, _, _) if !j.duplicateResolved =>
-        j.copy(right = dedupRight(left, right))
+      case j @ Join(left, right, _, condition) if !j.duplicateResolved =>
+        val (dedupedRight, attributeRewrites) = dedupRight(left, right)
+        val changedCondition = condition.map(_.transform {
+          case attr: Attribute if attr.resolved => attributeRewrites.getOrElse(attr, attr)
```

Comment:
This looks fishy. We will not rewrite attributes in the join condition when they are not resolved? What will happen then?

Reply:
Simply they will be resolved later.
`attributeRewrites` returned by `dedupRight()` can contain resolved attributes only.

Reply:
but actually there might be an issue here, instead of `attributeRewrites.getOrElse(attr, attr)` I might need to use `dedupAttr(attr, attributeRewrites)` which retains attribute qualifier

Reply:
changed to `dedupAttr()`